geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/FeatureSpecificReader.scala

```diff
@@ -21,22 +21,22 @@ import java.util.{Date, UUID}
 
 import com.vividsolutions.jts.geom.Geometry
 import org.apache.avro.Schema
-import org.apache.avro.io.{BinaryDecoder, DatumReader, Decoder, DecoderFactory}
+import org.apache.avro.io._
 import org.geotools.data.DataUtilities
 import org.geotools.filter.identity.FeatureIdImpl
 import org.locationtech.geomesa.feature.AvroSimpleFeatureUtils._
+import org.locationtech.geomesa.feature.EncodingOption.EncodingOptions
 import org.locationtech.geomesa.feature.serde.{ASFDeserializer, Version1Deserializer, Version2Deserializer}
+import org.locationtech.geomesa.feature.serialization.avro.AvroSimpleFeatureDecodingsCache
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
 import scala.collection.JavaConversions._
 
-class FeatureSpecificReader(oldType: SimpleFeatureType, newType: SimpleFeatureType)
-  extends DatumReader[AvroSimpleFeature] {
+class FeatureSpecificReader(oldType: SimpleFeatureType, newType: SimpleFeatureType, opts: EncodingOptions = EncodingOptions.none) {
```

Comment:
In general, we use the github line length, which corresponds to about 110 characters - you can set this up in intellij under editor -> code style